### PR TITLE
Fire field variable change event with variable value, not name.

### PR DESCRIPTION
**Pull request by marisaleung**
Prior, change events for field_variables were sending variables names. We have switched to using id's so this commit sends the new value's variable id instead.

Plus tests.

First part of fix for https://github.com/LLK/scratch-vm/issues/628

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly/1196)

### CODE REVIEW

File: core/field_variable.js [hunk 1]
```diff
@@ -103,30 +103,30 @@ Blockly.FieldVariable.prototype.getValue = function() {
...
     //     instances to use id instead of name.
-    else if (variable = this.sourceBlock_.workspace.getVariable(newValue)) {
-      this.setText(newValue);
-      this.value_ = variable.getId();
-      return;
+    else if (variable = this.sourceBlock_.workspace.getVariable(value)) {
+      // Value was a valid variable name.
+      newValue = variable.getId();
+    }
+    if (Blockly.Events.isEnabled()) {
```
Comment (rachel-fenichel) [inline core/field_variable.js]: At this point newValue is always an id, right?

Comment (marisaleung) [inline core/field_variable.js]: Not quite, I just added a few fixes to a couple of places that used name to now use id. But Blockly.Block.renameVar() still uses it. This seemed like a bigger change that I'd rather do in a separate CL.

Comment (marisaleung) [inline core/field_variable.js]: https://github.com/google/blockly/issues/1199
